### DISCUSSION TITLE
python only examples

### DISCUSSION THREAD

**Discussion by Harrymilo** [pywr/pywr#1139] [General]
Hi Thanks for the repository and the examples within it they are very useful. 

I'm using pyWR but would prefer to do all the coding within python as its the language I'm more familiar with. The examples provided (specifically the two reservoir system with optimisation and the thames example) are both coded in JSON and then called to python. 

Could you provide these examples coded solely within python? or else is there a functionality to write out the interpretation of the JSON code in python so that i can understand it better (and lift out sections of it)? 

Many thanks
Harry

**Comment by jetuk** [comment]
> Could you provide these examples coded solely within python? 

I don't currently have time to do that. There are more examples of using the Python API in the tests. 

> is there a functionality to write out the interpretation of the JSON code in python so that i can understand it better (and lift out sections of it)?

There is not. Largely the "simple" arguments are just passed as deserialised from JSON to the class constructors Python. The complication comes with passing nodes and parameters. These are typically defined as references (i.e. a name) in JSON but are passed as the loaded instance/object to the constructors in Python. I do not think the conversion would be too difficult.

**Reply by Harrymilo** [reply]
understood, thank you for the response
